R/Attributes.R

```diff
@@ -586,10 +586,67 @@ compileAttributes <- function(pkgdir = ".", verbose = getOption("verbose")) {
     list(env = list(PKG_CXXFLAGS ="-std=c++2b"))
 }
 
+.openmpPluginDefault <- function() {
+    list(env = list(PKG_CXXFLAGS = "-fopenmp", PKG_LIBS = "-fopenmp"))
+}
+
+.openmpPluginDarwin <- function() {
+
+    # get the C++ compiler from R
+    r <- file.path(R.home("bin"), "R")
+    cxx <- tryCatch(
+        system2(r, c("CMD", "config", "CXX"), stdout = TRUE),
+        condition = identity
+    )
+    if (inherits(cxx, "condition"))
+        return(.openmpPluginDefault())
+
+    # check the compiler version
+    command <- paste(cxx, "--version")
+    version <- tryCatch(
+        system(command, intern = TRUE),
+        condition = identity
+    )
+    if (inherits(version, "condition"))
+        return(.openmpPluginDefault())
+
+    # if we're using Apple clang, use alternate flags
+    # assume libomp was installed following https://mac.r-project.org/openmp/
+    if (any(grepl("Apple clang", version))) {
+
+        cxxflags <- "-Xclang -fopenmp"
+        libs <- "-lomp"
+
+    }
+
+    # if we're using Homebrew clang, add in libomp include paths
+    else if (any(grepl("Homebrew clang", version))) {
+
+        # ensure Homebrew paths are included during compilation
+        machine <- Sys.info()[["machine"]]
+        prefix <- if (machine == "arm64") "/opt/homebrew" else "/usr/local"
+
+        # build flags
+        cxxflags <- sprintf("-I%s/opt/libomp/include -fopenmp", prefix)
```

Comment:
I confirmed that the version of LLVM clang provided in Homebrew accepts `-fopenmp` as-is, but I'm adding the library paths for `libomp` just in case.